### ISSUE TITLE
Cannot use RGW overview in web dashboard

### ISSUE THREAD

**Issue by netthier** [rook/rook#11169]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When clicking on the "Object Gateway" section of the Ceph dashboard, I get the error "Page not found", and a small window at the top-right indicating a 500 Internal server error.  
Interestingly, the overview of the dashboard correctly shows "3 total" in the "Object Gateways" box.
**Expected behavior:**
I'd expect to see the CephObjectStore I deployed and be able to further configure it.
**How to reproduce it (minimal and precise):**

**File(s) to submit**:
*Object Store CR*
```yaml
apiVersion: ceph.rook.io/v1
kind: CephObjectStore
metadata:
  name: ceph-s3
  namespace: rook-ceph
spec:
  metadataPool:
    failureDomain: host
    replicated:
      size: 3
  dataPool:
    failureDomain: host
    erasureCoded:
      dataChunks: 2
      codingChunks: 1
  preservePoolsOnDelete: true
  gateway:
    port: 80
    instances: 3
  healthCheck:
    bucket:
      disabled: false
      interval: 60s
```
**Logs to submit**:
* Manager's logs*
```
debug 2022-10-18T15:32:38.289+0000 7f21764f2700  0 [dashboard ERROR rgw_client] Unable to create rgw user "dashboard": unable to get monitor info from DNS SRV with service name: ceph-mon
2022-10-18T15:32:38.289+0000 7f17c22d8500 -1 failed for service _ceph-mon._tcp
2022-10-18T15:32:38.289+0000 7f17c22d8500 -1 monclient: get_monmap_and_config cannot identify monitors to contact
failed to fetch mon config (--no-mon-config to skip)

debug 2022-10-18T15:32:38.289+0000 7f21764f2700  0 [dashboard ERROR rgw_client]
Traceback (most recent call last):
  File "/usr/share/ceph/mgr/dashboard/services/rgw_client.py", line 262, in configure_rgw_credentials
    assert access_key and secret_key
AssertionError
debug 2022-10-18T15:32:38.293+0000 7f21764f2700  0 [dashboard ERROR exception] Internal Server Error
Traceback (most recent call last):
  File "/usr/share/ceph/mgr/dashboard/services/rgw_client.py", line 262, in configure_rgw_credentials
    assert access_key and secret_key
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/ceph/mgr/dashboard/services/exception.py", line 47, in dashboard_exception_handler
    return handler(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/share/ceph/mgr/dashboard/controllers/_base_controller.py", line 258, in inner
    ret = func(*args, **kwargs)
  File "/usr/share/ceph/mgr/dashboard/controllers/rgw.py", line 54, in status
    instance = RgwClient.admin_instance()
  File "/usr/share/ceph/mgr/dashboard/services/rgw_client.py", line 367, in admin_instance
    return RgwClient.instance(daemon_name=daemon_name)
  File "/usr/share/ceph/mgr/dashboard/services/rgw_client.py", line 329, in instance
    configure_rgw_credentials()
  File "/usr/share/ceph/mgr/dashboard/services/rgw_client.py", line 267, in configure_rgw_credentials
    raise NoCredentialsException
dashboard.services.rgw_client.NoCredentialsException: No RGW credentials found, please consult the documentation on how to enable RGW for the dashboard.
debug 2022-10-18T15:32:38.293+0000 7f21764f2700  0 [dashboard ERROR request] [::ffff:10.244.1.7:38844] [GET] [500] [0.123s] [admin] [513.0B] /api/rgw/status
debug 2022-10-18T15:32:38.293+0000 7f21764f2700  0 [dashboard ERROR request] [b'{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "57d0bbe2-c62e-4928-b516-ac03781fd519"}
```
Notably, the referenced `ceph-mon` service does not exist, instead there are `rook-ceph-mon-a`, `rook-ceph-mon-b` etc. services.
**Cluster Status to submit**:
Healthy, nothing special.
**Environment**:
* OS (e.g. from /etc/os-release): Talos Linux 1.2.5
* Kernel (e.g. `uname -a`): 5.15.72-talos
* Cloud provider or hardware configuration: Hetzner AX41 servers
* Rook version (use `rook version` inside of a Rook Pod): v1.10.3
* Storage backend version (e.g. for ceph do `ceph -v`): 17.2.3
* Kubernetes version (use `kubectl version`): 1.25.0
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Talos
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)): Healthy

![image](https://user-images.githubusercontent.com/66856670/196478551-daf61362-d2d5-4baa-b145-410ca59e6920.png)
![image](https://user-images.githubusercontent.com/66856670/196478667-216cfb9d-1b81-4224-94c2-1bb6a87528bd.png)

**Comment by netthier** [comment]
Re-creating the `CephObjectStore` object seems to have fixed this.

**Comment by netthier** [comment]
Experiencing this again with a different object gateway, this time I dont have the possibility of recreating it since it already contains data. 
Errors are exactly the same

**Comment by travisn** [comment]
Shall we track with #9660? It looks similar.

**Comment by bumarcell** [comment]
This issue is different and shows a different error message
In this issue for example `ceph dashboard set-rgw-credentials` doesn't work because `Unable to create rgw user "dashboard": unable to get monitor info from DNS SRV with service name: ceph-mon` which doesn't exist

Funnily enough I have 3 production rook ceph clusters and each is throwing a different error when it comes to viewing RGW on the dashboard 😓

**Comment by travisn** [comment]
@bumarcell Have you tested with v18.2.1? There have been some rgw dashboard fixes in that release.

**Comment by GiamBoscaro** [comment]
Good morning,
I have the same problem on my dashboard now. I have configured a new cluster and everything works, but the Object Gateway pages. The errors I have is:

`{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "95d4d787-3649-4c51-bdc0-c14da1c1b2e5"} `

I have already tried all the tricks show in this thread here: https://github.com/rook/rook/issues/3026

**Comment by rkachach** [comment]
> Good morning,
> 
> I have the same problem on my dashboard now. I have configured a new cluster and everything works, but the Object Gateway pages. From the browser console I just get this:
> 
> `{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "95d4d787-3649-4c51-bdc0-c14da1c1b2e5"} `
> 
> From the manager logs I have:
> 
> ```
> 2024-02-23T08:55:56.104+0000 7fb46a7ea6c0  0 [dashboard ERROR exception] Internal Server Error
> Traceback (most recent call last):
>   File "/usr/share/ceph/mgr/dashboard/services/exception.py", line 47, in dashboard_exception_handler
>     return handler(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3/dist-packages/cherrypy/_cpdispatch.py", line 54, in __call__
>     return self.callable(*self.args, **self.kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/share/ceph/mgr/dashboard/controllers/_base_controller.py", line 263, in inner
>     ret = func(*args, **kwargs)
>           ^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/share/ceph/mgr/dashboard/controllers/_rest_controller.py", line 191, in wrapper
>     return func(*vpath, **params)
>            ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/share/ceph/mgr/dashboard/controllers/rgw.py", line 147, in list
>     'port': int(re.findall(r'port=(\d+)', metadata['frontend_config#0'])[0])
>                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
> IndexError: list index out of range
> 2024-02-23T08:55:56.104+0000 7fb46a7ea6c0  0 [dashboard ERROR request] [::ffff:10.20.111.119:41124] [GET] [500] [0.005s] [admin] [513.0B] /api/rgw/daemon
> 2024-02-23T08:55:56.104+0000 7fb46a7ea6c0  0 [dashboard ERROR request] [b'{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "ebbe31ed-d8e6-4998-99b7-a42f815b4465"}    
> ```
> 
> I have already tried all the tricks show in this thread here: #3026
> 
> EDIT: after having a look at the related python code, I have solved by specifying the port in the `ceph.conf:`
> 
> `rgw_frontends = "beast endpoint=10.20.110.121:8080 port=8080"`
> 
> Even if looking at the [documentation](https://docs.ceph.com/en/quincy/radosgw/frontends/) it shouldn't be specified when using `endpoint`.

@GiamBoscaro from the failing code I can see that you are not using the latest version of ceph. This issue was fixed by the PR https://github.com/ceph/ceph/pull/54531 which should be available on `v18.2.1`. However, this release has a regression (tracked by https://github.com/rook/rook/issues/13527) so meanwhile you can use the patched image proposed in the issue https://github.com/rook/rook/issues/13527. I'll recommend using that version as it also has fixes for other dashboard and rgw issues.
